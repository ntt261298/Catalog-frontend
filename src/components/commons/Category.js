import React, { Component } from 'react';
import { connect } from 'react-redux';
import { PropTypes } from 'prop-types';
import { Link } from 'react-router-dom';
import getCategories from '../../actions/category';

export class Category extends Component {
  componentDidMount() {
    this.props.getCategories();
  }

  render() {
    const { categories } = this.props;
    return (
      <div className="category">
        <h2>Catagories</h2>
        <ul>
          {
            categories.map(({ id, name }) => (
              <li key={id}>
                <Link to={`/category/${id}`}>{ name }</Link>
              </li>
            ))
          }
        </ul>
      </div>
    );
  }
}

Category.propTypes = {
  getCategories: PropTypes.func.isRequired,
  categories: PropTypes.array.isRequired,
};

const mapStateToProps = state => ({
  categories: state.category.categories,
});

export default connect(mapStateToProps, { getCategories })(Category);
