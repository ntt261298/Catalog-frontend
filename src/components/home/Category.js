import React, { Component } from 'react';
import { connect } from 'react-redux';
import { PropTypes } from 'prop-types';
import getCategories from '../../actions/category';

class Category extends Component {
  componentDidMount() {
    // eslint-disable-next-line react/destructuring-assignment
    this.props.getCategories();
  }

  render() {
    const { categories } = this.props;
    return (
      <div className="category">
        <h2>Catagories</h2>
        <ul>
          {
            categories.map(({ id, name }) => (
              <li key={id}>{ name }</li>
            ))
          }
        </ul>
      </div>
    );
  }
}

Category.propTypes = {
  getCategories: PropTypes.func.isRequired,
  categories: PropTypes.arrayOf.isRequired,
};

const mapStateToProps = state => ({
  categories: state.category.categories,
});

export default connect(mapStateToProps, { getCategories })(Category);
