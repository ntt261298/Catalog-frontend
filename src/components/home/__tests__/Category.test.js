import React from 'react';
import { shallow } from '../../../enzyme';
import { Category } from '../Category';

describe('Category tests', () => {
  let wrapper;
  let props;
  const setup = () => {
    wrapper = shallow(
      <Category {...props} />,
    );
  };

  beforeEach(() => {
    props = {
      ...props,
      categories: [{ id: 1, items: [], name: 'One' },
        { id: 1, items: [], name: 'Two' },
        { id: 1, items: [], name: 'Three' }],
      getCategories: () => props.categories,
    };
  });

  it('renders list-categories', () => {
    setup();
    // Expect the wrapper object to be defined
    expect(wrapper).toMatchSnapshot();
  });
});
