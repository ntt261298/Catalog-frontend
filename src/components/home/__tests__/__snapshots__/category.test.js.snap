// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Category tests renders list-categories 1`] = `
<div
  className="category"
>
  <h2>
    Catagories
  </h2>
  <ul>
    <li
      key="1"
    >
      One
    </li>
    <li
      key="1"
    >
      Two
    </li>
    <li
      key="1"
    >
      Three
    </li>
  </ul>
</div>
`;
